### ISSUE TITLE
How to train the model with binary cross-entropy?

### ISSUE THREAD

**Issue by qrshi-17** [pterhoer/FaceImageQuality#42]
How to divide the images in a dataset into high quality and low quality?

**Comment by qrshi-17** [comment]
> Hi qrshi-17,
> I think I am lacking context information to answer your question properly.
> 
> 1. Which model are you referring to? SER-FIQ is a concept that can be applied to arbitrary face recognition models.
> 2. The concept is training free.
> 3. If you want to use this method for creating labels to train your own model (with a cross-entropy loss), then you can apply SER-FIQ to your database. It will give you continuous quality values that can be divided into high and low quality by using a simple threshold. With this threshold, you can define what is high and low quality.
>    Best
>    Philipp

Hi Philipp,
Thank you for your patient reply.
I was trying to re-implement the training phase of the top-model mentioned in the paper. In the experiment setting, the backbone is freezed, and the fully connected layer is trained with the cross-entropy loss. I am doubt about how to provide the labels for training if I use colorferet dataset.
Looking forward to your reply.
Best
Qr

**Comment by pterhoer** [comment]
Hi Qr,

the "on-top" model is only a new face recognition (FR) model trained with dropout. It is trained to differentiate between the training identities and thus, the labels used are the training IDs. (No qualit labels are involved. How could we define quality anyway? :) )

The idea of using this strategy was to overcome the problem that you need a FR model trained with dropout.
If you have a FR network that is already trained with dropout I strongly recommend to apply SER-FIQ on this network since it will lead to better performances (the network is trained on the whole training data while the new layer might not).

Best
Philipp

**Comment by qrshi-17** [comment]
> Hi qrshi-17,
> I think I am lacking context information to answer your question properly.
> 
> 1. Which model are you referring to? SER-FIQ is a concept that can be applied to arbitrary face recognition models.
> 2. The concept is training free.
> 3. If you want to use this method for creating labels to train your own model (with a cross-entropy loss), then you can apply SER-FIQ to your database. It will give you continuous quality values that can be divided into high and low quality by using a simple threshold. With this threshold, you can define what is high and low quality.
>    Best
>    Philipp

Hi Philipp,
Thank you for your patient reply.
I was trying to re-implement the training phase of the top-model mentioned in the paper. In the experiment setting, the backbone is freezed, and the fully connected layer is trained with the cross-entropy loss. I am doubt about how to provide the labels for training if I use colorferet dataset.
Looking forward to your reply.
Best
Qr

> Hi Qr,
> 
> the "on-top" model is only a new face recognition (FR) model trained with dropout. It is trained to differentiate between the training identities and thus, the labels used are the training IDs. (No qualit labels are involved. How could we define quality anyway? :) )
> 
> The idea of using this strategy was to overcome the problem that you need a FR model trained with dropout.
> If you have a FR network that is already trained with dropout I strongly recommend to apply SER-FIQ on this network since it will lead to better performances (the network is trained on the whole training data while the new layer might not).
> 
> Best
> Philipp

I got it wrong😂, thanks for answering my doubts.
I have tried your method, it works well no my task. It's good idea.

Best
Qr

**Comment by pterhoer** [comment]
You are welcome :)
